### DISCUSSION TITLE
Build got stuck at notarizing the Mac app for more than an hour.

### DISCUSSION THREAD

**Discussion by billywatkins** [tauri-apps/tauri#8630] [Q&A] [answered]
Hey, I am in the final stage of finishing my app. Thanks for the write-up about [Code Signing on Mac](https://tauri.app/v1/guides/distribution/sign-macos/). I followed and set all the proper environment variables, but my build process takes way too long for notarizing it.

This is the point where the build got stuck.
```
Running [tauri_bundler::bundle::common] Command `xcrun  notarytool submit ....
```

When I checked my notarization log, it showed zero. I am confused; does this mean Tauri hasn't even started the process?

```
 *** Warning: altool has been deprecated for notarization and starting in late 2023 will no longer be supported by the Apple notary service. You should start using notarytool to notarize your software. 

Notarization History - page 0

No items found.
```

UPDATE:

Okay, my bad. I used xcrun notarytool history and got the list. It turns out all of them are in progress, and the oldest one has been more than 4 hours. Is this normal?

**Comment by FabianLars** [comment, marked as answer]
> Okay, my bad. I used xcrun notarytool history and got the list. It turns out all of them are in progress, and the oldest one has been more than 4 hours. Is this normal?

Sometimes yes. If none of your apps were ever notarized before and/or your apps are fairly large in size it may take a while. Especially considering that the notarization servers' performance is well, let's call it fluctuating...

**Reply by billywatkins** [reply]
Thank you!

**Reply by luis-cicada** [reply]
@billywatkins did at some point the notarizing process completes? I'm on the same boat

**Reply by habbes** [reply]
Also on the same boat. Did the notarization process eventually complete?

**Reply by luis-cicada** [reply]
> Also on the same boat. Did the notarization process eventually complete?

Yes it does. Just hang in there.

**Reply by habbes** [reply]
Thanks @luis-cicada, and did it take hours for every release after that or just the first one?

**Reply by luis-cicada** [reply]
> Thanks @luis-cicada, and did it take hours for every release after that or just the first one?

Just the first time.

**Reply by habbes** [reply]
Hey, @luis-cicada it worked finally. Thanks. Took 2 days but eventually worked, now it completes fairly quickly. I don't know whether running on the weekend had anything to do with it.

**Reply by silentcastledev** [reply]
Folks, is there a way to make notarization process emit some logs, any logs, please? When it takes long, it is not actually clear what is going on.

**Reply by luis-cicada** [reply]
> Folks, is there a way to make notarization process emit some logs, any logs, please? When it takes long, it is not actually clear what is going on.

You can alway use `--verbose `

**Reply by ngaer** [reply]
Also sometimes notary service experience outages, you can check it on the Apple system status page - https://developer.apple.com/system-status

<img width="1494" alt="Screenshot 2025-02-06 at 23 18 25" src="https://github.com/user-attachments/assets/f0c701b5-1df0-4d4d-a9c8-1ddaafec9e65" />

**Reply by oliverga** [reply]
Mine has been stuck on "In Progress" for 5 days. Has anyone else experienced this? Getting a little antsy here.

**Reply by oliverga** [reply]
> Mine has been stuck on "In Progress" for 5 days. Has anyone else experienced this? Getting a little antsy here.

Just got accepted now – truly is just a waiting game!

**Reply by willwatson-dev** [reply]
@oliverga Wow, that's ridiculous! Can I ask, is this your first time submitting for notarization? I am in the same boat, where my notarization submissions take hours and hours. I have added multiple logging functions in my pipeline to keep me up to date, but I am getting hung on In Progress for 5+ hours so far. This project is my first time using the notary service, and I've heard this happens to first time users. Is this the case for you as well? Thanks.

**Reply by luis-cicada** [reply]
@willwatson-bot this will happen only the first time. Once the notarization is completed the first time, all others will be super fast.

**Reply by ashutoshxhq** [reply]
I am doing my first notarization submission,  @oliverga  Did you keep it running for 5 days straight? or tried few hrs each day? I am loosing hope, exhausted my github actions. Do I have to run this on a server ?

**Reply by oliverga** [reply]
@ashutoshxhq I used [notarytool](https://developer.apple.com/documentation/security/notarizing-macos-software-before-distribution), and just waited. The process runs on their servers, so after submitting successfully you can exit the process and close the terminal with no issue. Then you can check on your submission status using ```xcrun notarytool history```

**Reply by willwatson-dev** [reply]
Just wanted to jump back in and let you all know that my submission was approved. @ashutoshxhq , I would second what @oliverga has recommended. You are able to poll the notary service to check in later. I develop on Windows, and so too I rely on GitHub Actions to bundle, sign, and send my application to notary. 

I would highly recommend building logging in to your workflow so that you can see what went wrong with your notarization (you can read more here: https://developer.apple.com/documentation/notaryapi/get-submission-log). You should get a URL to an S3 endpoint when you send a get request to them for the submission log. This is recommended in the Notary documentation. This submission log will tell you the status of your submission, and if it was rejected or invalid, it will you give a reason why. This is very useful for determining what might be going wrong.

In my case, it was an error that looked something like this: "Team is not yet configured for notarization. Please contact Developer Programs Support at developer.apple.com under the topic Development and Technical / Other Development or Technical Questions." and a status code of: 7000. Basically all of my submissions had been taking over 12 hours each, all getting rejected, but they all had this response when I retrieved the log for the submission. Basically I had to reach out to developer support to get this resolved, and now notary for the exact same code takes less than 4 minutes consistently.

@ashutoshxhq don't lose hope my friend! Try first adding visibility and logging in to your workflow with the submission log. If you are new to Notary, as I was, it seems that there is some sort of manual configuration that Apple needs to do on their end before they approve you going forward. In my particular case, I didn't make any headway until I reached out to customer support multiple times. It was resolved quickly once I was escalated to Developer support.

**Reply by willwatson-dev** [reply]
> Just wanted to jump back in and let you all know that my submission was approved. @ashutoshxhq , I would second what @oliverga has recommended. You are able to poll the notary service to check in later. I develop on Windows, and so too I rely on GitHub Actions to bundle, sign, and send my application to notary.
> 
> I would highly recommend building logging in to your workflow so that you can see what went wrong with your notarization (you can read more here: https://developer.apple.com/documentation/notaryapi/get-submission-log). You should get a URL to an S3 endpoint when you send a get request to them for the submission log. This is recommended in the Notary documentation. This submission log will tell you the status of your submission, and if it was rejected or invalid, it will you give a reason why. This is very useful for determining what might be going wrong.
> 
> In my case, it was an error that looked something like this: "Team is not yet configured for notarization. Please contact Developer Programs Support at developer.apple.com under the topic Development and Technical / Other Development or Technical Questions." and a status code of: 7000. Basically all of my submissions had been taking over 12 hours each, all getting rejected, but they all had this response when I retrieved the log for the submission. Basically I had to reach out to developer support to get this resolved, and now notary for the exact same code takes less than 4 minutes consistently.
> 
> @ashutoshxhq don't lose hope my friend! Try first adding visibility and logging in to your workflow with the submission log. If you are new to Notary, as I was, it seems that there is some sort of manual configuration that Apple needs to do on their end before they approve you going forward. In my particular case, I didn't make any headway until I reached out to customer support multiple times. It was resolved quickly once I was escalated to Developer support.

Sorry and just to be clear, it was an Apple-side issue. Sometimes they just need to approve you. You'll only know for sure if you check the submission log, however.

**Reply by ashutoshxhq** [reply]
thanks for the help @oliverga @willwatson-bot. My app just got approved, It was timing out initially but after few hours of retries, one of the tries went through. 
After the first one was approved, it has been very fast. 1-2 mins for each approval.

Thanks a bunch.

**Comment by yonson40** [comment]
guys, I've been stuck in the "In Progress" phase all day... This is my first time notarizing. I've pushed three attempts through with notary tool and one via electron notary. Any tips or just wait it out..

**Reply by willwatson-dev** [reply]
Hey there, I'm here to reassure you my friend! This is absolutely normal for your first time. This happened to all of us. What you'll want to do to really move things along is get in touch with developer support and let them know that you're having issues. Squeaky wheel gets the oil! I found that I was not getting anything to pass until I reached out to them. Now my builds take 3 min tops. All of mine before approval were taking 4 hours and then failing. I would highly recommend building logging in to your workflow so that you can see what went wrong with your notarization (you can read more here: https://developer.apple.com/documentation/notaryapi/get-submission-log). 

Hang in there, and again, follow visibility and logging best practices so you can really be sure of what is going wrong! Learning how to keep your bugs/issues/logs organized will keep your sanity for these things!!

**Reply by yonson40** [reply]
Thanks for the quick reply! I reached out to support earlier, will wait for them!

**Reply by skyline69** [reply]
@yonson40 Any updates on this so far?

**Reply by yonson40** [reply]
@skyline69 yes! As suggested, I waited it out until the net day, seems like the queue cleared up, and now my notarizations take a few minutes max!

**Reply by skyline69** [reply]
I notarized it on my workflow for the first time and it worked fine. But after that I was wating like 45mins and nothing happened since in the future actions. Does that happen for you too on your workflows?

**Reply by Burpbounty** [reply]
> I notarized it on my workflow for the first time and it worked fine. But after that I was wating like 45mins and nothing happened since in the future actions. Does that happen for you too on your workflows?我第一次在我的工作流程中对其进行了公证，效果很好。但在那之后，我等待了大约 45 分钟，从那以后的行动中什么也没发生。您的工作流程中也会发生这种情况吗？

I’ve experienced the same situation as you and shared it on Twitter. Please follow and support us! Let’s fight for the rights we deserve! https://x.com/silent_orchid6/status/1954399957479567603

**Comment by Burpbounty** [comment]
> Hey, I am in the final stage of finishing my app. Thanks for the write-up about [Code Signing on Mac](https://tauri.app/v1/guides/distribution/sign-macos/). I followed and set all the proper environment variables, but my build process takes way too long for notarizing it.嘿，我正处于完成我的应用程序的最后阶段。感谢您关于 [Mac 上代码签名](https://tauri.app/v1/guides/distribution/sign-macos/)的文章。我遵循并设置了所有正确的环境变量，但我的构建过程需要很长时间来公证它。
> 
> This is the point where the build got stuck.这是构建卡住的地方。
> 
> ```
> Running [tauri_bundler::bundle::common] Command `xcrun  notarytool submit ....
> ```
> 
> When I checked my notarization log, it showed zero. I am confused; does this mean Tauri hasn't even started the process?当我检查我的公证日志时，它显示为零。我很困惑;这是否意味着金牛座甚至还没有开始这个过程？
> 
> ```
>  *** Warning: altool has been deprecated for notarization and starting in late 2023 will no longer be supported by the Apple notary service. You should start using notarytool to notarize your software. 
> 
> Notarization History - page 0
> 
> No items found.
> ```
> 
> UPDATE:  更新：
> 
> Okay, my bad. I used xcrun notarytool history and got the list. It turns out all of them are in progress, and the oldest one has been more than 4 hours. Is this normal?好吧，我的坏。我使用了 xcrun notarytool 历史记录并获得了列表。原来都在进行中，最旧的已经 4 个多小时了。这正常吗？

I’ve experienced the same situation as you and shared it on Twitter. Please follow and support us! Let’s fight for the rights we deserve! https://x.com/silent_orchid6/status/1954399957479567603
